test/Microsoft.AspNetCore.Routing.Tests/RouteValueBasedEndpointFinderTest.cs

```diff
@@ -169,6 +170,31 @@ public void EndpointDataSource_ChangeCallback_Refreshes_OutboundMatches()
                 });
         }
 
+        [Fact]
+        public void FindEndpoints_ReturnsEndpoint_WhenLookedUpByRouteName()
+        {
+            // Arrange
+            var expected = CreateEndpoint(
+                "api/orders/{id}",
+                defaults: new { controller = "Orders", action = "GetById" },
+                requiredValues: new { controller = "Orders", action = "GetById" },
+                routeName: "OrdersApi");
+            var finder = CreateEndpointFinder(expected);
+
+            // Act
+            var foundEndpoints = finder.FindEndpoints(
+                new RouteValuesBasedEndpointFinderContext
+                {
+                    ExplicitValues = new RouteValueDictionary(new { id = 10 }),
+                    AmbientValues = new RouteValueDictionary(new { controller = "Home", action = "Index" }),
+                    RouteName = "OrdersApi"
+                });
+
+            // Assert
+            var actual = Assert.Single(foundEndpoints);
+            Assert.Same(expected, actual);
+        }
```

Comment:
Do we have a test for the failure case? Like this, but without values for controller/action

Reply:
I am a bit unclear @rynowak . There are no explicit values for controller/action in the above test actually.

Reply:
I am planning to add more tests to GlobalRoutingUrlHelperTest in MVC